java/server/src/org/openqa/selenium/grid/sessionqueue/NewSessionQueue.java

```diff
@@ -44,7 +44,7 @@
 
   public abstract boolean offerFirst(HttpRequest request, RequestId requestId);
 
-  public abstract Optional<HttpRequest> poll();
+  public abstract Optional<HttpRequest> poll(RequestId requestId);
```

Comment:
The expected behaviour for `poll` would be to match what `Deque` does, which is: 

> Retrieves and removes the head of the queue represented by this deque

As such, you shouldn't need the `RequestId` here.